src/System.Net.Http/src/System/Net/Http/Managed/HttpEnvironmentProxy.cs

```diff
@@ -0,0 +1,265 @@
+            if (httpProxy == null && httpsProxy == null)
+            {
+                return null;
+            }
+
+            return new HttpEnvironmentProxy(httpProxy, httpsProxy, Environment.GetEnvironmentVariable(envNoProxyLC));
+        }
+
+        private HttpEnvironmentProxy(Uri httpProxy, Uri httpsProxy, string bypassList)
+        {
+            NetworkCredential httpCred = null;
+            NetworkCredential httpsCred = null;
+            bool sameAuth = false;
+
+            if (httpProxy != null)
+            {
+                _http = httpProxy;
+                httpCred = GetCredientialsFromString(_http.UserInfo);
+            }
+            if (httpsProxy != null)
+            {
+                _https = httpsProxy;
+                httpsCred = GetCredientialsFromString(_https.UserInfo);
+                if ((_http != null) && (_http.UserInfo == _https.UserInfo))
+                {
+                    sameAuth = true;
+                }
+            }
+            if (httpCred != null || httpsCred != null)
+            {
+                // If only one protocol is set or both protocols have same auth,
+                // use standard credential object
+                if (sameAuth || httpCred == null || httpsCred == null)
+                {
+                    _credientials = httpCred ?? httpsCred;
+                }
+                else
+                {
+                    // use wrapper so we can decide later based on uri
+                    _credientials = new HttpEnvironmentProxyCredientials(httpCred, httpsCred);
+                }
+            }
+
+            if (!string.IsNullOrWhiteSpace(bypassList))
+            {
+                string[] list = bypassList.Split(',');
+                List<string> tmpList = new List<string>();
+
+                foreach (string value in list)
+                {
+                    string tmp = value.Trim();
+                    if (tmp.Length > 0)
+                    {
+                        tmpList.Add(tmp);
+                    }
+                }
+                if (tmpList.Count > 0)
+                {
+                    _bypass = tmpList.ToArray();
+                }
+            }
+        }
+
+        /// <summary>
+        /// This function will evaluate given string and it will try to convert
+        /// it to Uri object. The string could contain URI fragment, IP address and  port
+        /// tuple or just IP address or name. It will return null if parsing fails.
+        /// </summary>
+        private static Uri GetUriFromString(String value)
+        {
+            if (value == null)
+            {
+                return null;
+            }
+
+            Uri uri = null;
+            try
+            {
+                uri = new Uri(value);
+            }
+            catch (UriFormatException)
+            {
+                if (value.Contains("://"))
+                {
+                    throw;
```

Comment:
This exception should be thrown as an HttpRequestException, with appropriate error message..